src/TapeRoller.ts

```diff
@@ -0,0 +1,253 @@
+    const fullPath = `${path}/${fileName}`;
+
+    mkdirp.sync(this.targetDirectory);
+
+    this.stream = this.stream.pipe(fs.createWriteStream(fullPath));
+
+    this.stream.on('error', (error) => console.error('Write error: ', error));
+
+    if (temp) {
+      this.stream.on('close', () => {
+        this.rename(fullPath, `${path}/${this.targetFile}`);
+      });
+    }
+
+    return this;
+  }
+
+  public copy (source?: string, target?: string): this {
+    if (source) {
+      this.sourceFile = source;
+    }
+
+    if (target) {
+      this.targetFile = target;
+    }
+
+    if (!this.stream) {
+      this.read();
+    }
+
+    this.write();
+
+    return this;
+  }
+
+  public replace (parameters: { [key: string]: any }) {
+    if (!this.stream) {
+      this.read();
+    }
+
+    const params  = new Homefront(parameters);
+    const regex   = /{{\s?([\w.]+)(?:(?:\s?:\s?)(?:(?:['"]?)(.*?)(?:['"]?)))?\s?}}/gi;
+    const replace = (match: string, parameter: string, defaultValue: string): string => {
+      return params.fetch(parameter) || defaultValue || match;
+    };
+
+    this.stream = this.stream.pipe(this.applyModify(regex, { custom: replace }));
+
+    return this;
+  }
+
+  public modify (pattern: RegExp, { append, prepend, custom, replace }: ModifyType): this {
+    if (!this.stream) {
+      this.read();
+    }
+
+    this.stream = this.stream.pipe(this.applyModify(pattern, { append, prepend, custom, replace }));
+
+    return this;
+  }
+
+  private applyModify (pattern: RegExp, { append, prepend, custom, replace }: ModifyType): fs.WriteStream {
+    return streamReplace(pattern, (match: string, parameter: string, defaultValue: string): string => {
+      if (typeof replace !== 'undefined') {
+        return replace;
+      }
+
+      if (typeof append !== 'undefined') {
+        return match + append;
+      }
+
+      if (typeof prepend !== 'undefined') {
+        return prepend + match;
+      }
+
+      if (typeof custom === 'function') {
+        return custom(match, parameter, defaultValue);
+      }
+
+      return match;
+    });
+  }
+
+  public clone (repository: string, projectName?: string, cwd: string = null) {
+    const args = ['clone', repository];
+
+    if (projectName) {
+      args.push(projectName);
+    }
+
+    const process = spawn('git', args, { cwd });
+    const name    = projectName || repository.match(/\/([\w-_]+)\.git$/)[1];
+    const path    = cwd ? `./${cwd}/${name}` : `./${name}`;
+
+    process.on('close', (status: number) => {
+      if (status !== 0) {
+        return;
+      }
+
+      this.remove(`${path}/.git`, true);
+      this.installDependencies(path);
+    });
+
+    process.on('error', (error: { code: string }) => {
+      console.log('Could not clone repository ' + repository);
+    });
+
+    return this;
+  }
+
+  private installDependencies (path: string): this {
+    const process = exec(`yarn --cwd ${path} install`);
```

Comment:
How do you know the user uses yarn? https://github.com/SpoonX/boards-preset-default/blob/master/lib/task/prepare.js#L3

Reply:
Black magic 🔮

Reply:
:D